napari/_qt/qt_viewer.py

```diff
@@ -248,6 +249,25 @@ def _open_folder(self):
         if folder not in {'', None}:
             self._add_files([folder])
 
+    def _add_path(self, path):
+        """Add filename, directory, or URL to the viewer.
+
+        Will look for first plugin implementing napari_get_reader that
+        recognizes ``path`` as a compatible format, and then use it to add
+        data to the viewer.
+
+        Parameters
+        -------
+        paths : str
+            path to filename, directory, or URL to add to the viewer.
+        """
+        if path:
+            reader = plugin_manager.hook.napari_get_reader(path=path)
+            layer_data = reader(path)
+            for data in layer_data:
+                self.viewer._add_layer_data(*data)
+            self._last_visited_dir = os.path.dirname(path)
+
     def _add_files(self, filenames):
```

Comment:
what's your plan for `_add_files` vs `_add_paths`, can we drop `_add_files` in this PR too? I think drag / drop and file open should always do the same thing and we may only need one of those methods (also I totally get that this is still in progress, I'm just taking a look now and trying to be helpful)

Reply:
Yes, I did intend to remove `_add_files`, but wanted to check with you first.  there's a fair amount of logic in here that is ultimately aimed at our `magic_imread` function which accepts a list of directories in order to make pyramids (right?).  so we'll need to decide how to differentiate between a multi-file drop intended for pyramids, vs a multi-file open intended for multiple layers.  I think it would significantly simplify things if everything dropped (or handed to `File > Open`) was treated as an independent entity (whether it be a file, directory, or even URL to remote resource) that the readers get a crack at.  And multi-file pyramid opening should live in a `File > Open Pyramid` or something like that.

However... let me know if am missing some of the original intention of the `filenames` as a list instead of a single file.

Reply:
> magic_imread function which accepts a list of directories in order to make pyramids (right?)

Not quite - I think this was intended by @jni for creating a stack where the first dimension corresponds to the file ID (i.e. imagine you have a timeseries of 2D images where each file is a timepoint and you want to then view them as a single dim-3 array with a slider for time).

I'm not quite sure about the list of files - maybe that's something that a reader plugin should accept too - regardless, standardizing all of this now and encouraging us to use plugin reader functionality and syntax internally as much as possible all sound good to me

Reply:
hmm, i guess we have to make some assumptions one way or the other.
in my own data collection, usually if I have a folder of 2D images representing 3D (z/t) data, I want to just open the whole folder, rather than selecting all the files and passing them to the program... but both use-cases will undoubtedly arise, so we should just lay out all the possibilities of file-opening, and decide what we're going to do with each case.  ill follow up with a comment on that later

Reply:
actually... I'm going to just remove all of the code that actually connects the reader plugin hook from this PR, and will open a new PR to discuss how best to plug in the hook.  This PR can then be dedicated just to adding a pluggy.PluginManager.  I'll link this comment thread in that PR when I open it

Reply:
Sounds good - keeping this PR as small as possible - no IO / reader stuff sounds good